microcosm/loaders.py

```diff
@@ -109,7 +109,10 @@ def matches_key(key_parts):
             dct[key_part.lower()] = dict()
             dct = dct[key_part.lower()]
         # set the value for the final part
-        dct[key_parts[-1].lower()] = value
+        try:
+            dct[key_parts[-1].lower()] = loads(value)
```

Comment:
This requires that the envvar value be quoted, which I think is an ok compromise.

```
FOO=bar
```
will be interpreted as a string

```
FOO="bar"
```

will be interpreted as a JSON string.

This makes it harder to have a literal quoted strings as values, but I'm not sure we need that.

The alternative is to have a different loader for JSON. Thoughts @adamhadani ?